### PR TITLE
En 4158 metablocks refactor

### PR DESCRIPTION
**Pull request by iulianpascalau**
refactored metablock interceptor
removed peer changes interceptor

### CODE REVIEW

File: process/block/interceptedBlocks/interceptedMetaBlockHeader.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+	inHdr := &InterceptedMetaHeader{
+		hdr:                 hdr,
+		marshalizer:         arg.Marshalizer,
+		hasher:              arg.Hasher,
+		multiSigVerifier:    arg.MultiSigVerifier,
+		chronologyValidator: arg.ChronologyValidator,
+		shardCoordinator:    arg.ShardCoordinator,
+	}
+	inHdr.processFields(arg.HdrBuff)
```
Comment (miiu96) [inline process/block/interceptedBlocks/interceptedMetaBlockHeader.go]: Maybe you can store in arg.HrdBuff hashed HdrBuff and remove processFields from constructor method.

File: process/interceptors/factory/metaInterceptedDataFactory_test.go [hunk 2]
```diff
@@ -126,6 +126,23 @@ func TestMetaInterceptedDataFactory_CreateInterceptedShardHdrShouldWork(t *testi
...
 
+func TestMetaInterceptedDataFactory_CreateInterceptedMetaHdrShouldWork(t *testing.T) {
+	t.Parallel()
+
+	marshalizer := &mock.MarshalizerMock{}
+	emptyHdr := &block.Header{}
+	emptyHdrBuff, _ := marshalizer.Marshal(emptyHdr)
+
+	midf, _ := factory.NewMetaInterceptedDataFactory(createMockMetaArgument(), factory.InterceptedMetaHeader)
+
```
Comment (miiu96) [inline process/interceptors/factory/metaInterceptedDataFactory_test.go]: remove empty line.

File: process/interceptors/factory/shardInterceptedDataFactory_test.go [hunk 3]
```diff
@@ -234,6 +234,23 @@ func TestShardInterceptedDataFactory_CreateInterceptedShardHdrShouldWork(t *test
...
 
+func TestShardInterceptedDataFactory_CreateInterceptedMetaHdrShouldWork(t *testing.T) {
+	t.Parallel()
+
+	marshalizer := &mock.MarshalizerMock{}
+	emptyHdr := &block.Header{}
+	emptyHdrBuff, _ := marshalizer.Marshal(emptyHdr)
+
+	midf, _ := factory.NewShardInterceptedDataFactory(createMockShardArgument(), factory.InterceptedMetaHeader)
+
```
Comment (miiu96) [inline process/interceptors/factory/shardInterceptedDataFactory_test.go]: remove empty line.

File: process/block/interceptedBlocks/common.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+			return err
+		}
+	}
+
+	return nil
+}
+
+func checkShardData(sd block.ShardData, coordinator sharding.Coordinator) error {
+	for _, smbh := range sd.ShardMiniBlockHeaders {
+		if smbh.SenderShardId >= coordinator.NumberOfShards() {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply process/block/interceptedBlocks/common.go]: Done

File: process/block/interceptedBlocks/common.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+			return process.ErrInvalidShardId
+		}
+	}
+
+	return nil
+}
+
+func checkMiniblocks(miniblocks []block.MiniBlockHeader, coordinator sharding.Coordinator) error {
+	for _, miniblock := range miniblocks {
+		if miniblock.SenderShardID >= coordinator.NumberOfShards() {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply process/block/interceptedBlocks/common.go]: Done

File: process/block/interceptedBlocks/common_test.go [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+func createDefaultArgument() *ArgInterceptedBlockHeader {
+	arg := &ArgInterceptedBlockHeader{
+		ShardCoordinator: mock.NewOneShardCoordinatorMock(),
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply process/block/interceptedBlocks/common_test.go]: Reformatted

File: process/interceptors/factory/metaInterceptedDataFactory_test.go (hunk 2 above)
Comment (iulianpascalau) [reply process/interceptors/factory/metaInterceptedDataFactory_test.go]: Removed line 135 as to respect the 3 main parts of a test func: arrange, act and assert

File: process/interceptors/factory/shardInterceptedDataFactory_test.go (hunk 3 above)
Comment (iulianpascalau) [reply process/interceptors/factory/shardInterceptedDataFactory_test.go]: Removed or added on the whole test files as to conform to AAA

**Pull request merged**